lib/zendesk_apps_support/validations/mime.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module ZendeskAppsSupport
+  module Validations
+    module Mime
+      UNSUPPORTED_MIME_TYPES = %w[
+        vnd.rar rar zip gzip pdf doc docx avi bin bz bz2 csh sh jar mp3 mpeg odt pptx ppt xls xlsx 7z
+      ].freeze
+
+      class << self
+        def call(package)
+          unsupported_files = package.files.find_all do |app_file|
+            identify_mime_type(app_file) && block_listed?(app_file)
+          end.map(&:relative_path)
+
+          [mime_type_warning(unsupported_files)] if unsupported_files.any?
+        end
+
+        private
+
+        def block_listed?(app_file)
+          app_file_mime_type = identify_mime_type(app_file)
```

Comment:
nit: could just call this variable `mime_type` and save some reading

Reply:
YES! I like!